sys/include/net/ng_pktbuf.h

```diff
@@ -156,12 +158,27 @@ void ng_pktbuf_release(ng_pktsnip_t *pkt);
  * @param[in] pkt   The packet you want to write into.
  *
  * @return  The (new) pointer to the pkt.
- * @return  NULL, if ng_pktsnip_t::users of @p pkt > 1 and if there is not enough
- *          space in the packet buffer.
+ * @return  NULL, if ng_pktsnip_t::users of @p pkt > 1 and if there is not
+ *          enough space in the packet buffer.
  */
 ng_pktsnip_t *ng_pktbuf_start_write(ng_pktsnip_t *pkt);
 
 /**
+ * @brief   Create a IOVEC representation of the packet pointed to by *pkt*
+ *
+ * @details This function will create a new packet snip in the packet buffer,
+ *          which points to the given *pkt* and contains a IOVEC representation
+ *          of the referenced packet in its data section.
+ *
+ * @param[in]  pkt  Packet to export as IOVEC
+ * @param[out] len  Number of elements in the IOVEC
+ *
+ * @return  Pointer to the 'IOVEC packet snip'
+ * @return  NULL, if packet is empty of the packet buffer is full
+ */
+ng_pktsnip_t *ng_pktbuf_get_iovec(ng_pktsnip_t *pkt, size_t *len);
```

Comment:
Out-parameters first, please. And out of pure practicality: Wouldn't it make more sense to make the `size_t` to be the return value and the IOVEC `ng_pktsnip_t` to be the out parameter?